numpy/core/src/multiarray/lowlevel_strided_loops.c.src

```diff
@@ -1707,10 +1712,22 @@ mapiter_@name@(PyArrayMapIterObject *mit)
                 }
 
                 /*
-                 * Resetting is slow, so skip if the subspace iteration has
-                 * only a single inner loop.
+                 * Resetting is slow, so skip if the subspace iteration is
```

Comment:
@seberg I am not sure if the prediction works correctly for higher dimensional subspace.
Saying the fastest dimension has strides == +-itemsize, but the last-fastest dimension doesn't. Not sure if I can quickly come up with a test case.